prov/rxm/src/rxm.h

```diff
@@ -685,6 +685,23 @@ rxm_ep_handle_unconnected(struct rxm_ep *rxm_ep, struct util_cmap_handle *handle
 	return -FI_EAGAIN;
 }
 
+static inline int
+rxm_acquire_conn_connect(struct rxm_ep *rxm_ep, fi_addr_t fi_addr,
+			 struct rxm_conn **rxm_conn)
+{
+	*rxm_conn = rxm_acquire_conn(rxm_ep, fi_addr);
```

Comment:
Are locks not required around this call?

Reply:
I think it should be
```
*rxm_conn = rxm_acquire_conn(rxm_ep, fi_addr);
if (!*rxm_conn) {
    acquire(cmap::lock);
    if (!*rxm_conn || ((*rxm_conn)->handle.state != CMAP_CONNECTED)) {
          /* connect */
    }
    release(cmap::lock)
}
```
It shouldn't be racy in this case.

Reply:
@a-ilango -- any comment on this?  I can look deeper into the code if needed to see if there's an actual race.  I believe Dmitry is on vacation.

Reply:
On further inspection, `rxm_acquire_conn` should always return a valid `rxm_conn`. Otherwise it's an application error (address not in AV) or a bug. I guess we should separate that check. Other than that I think the code in the patch is fine. When `handle.state == CMAP_CONNECTED`, we should have a valid msg_ep.

Reply:
Okay - got it.  So is this patch good to merge then.

Reply:
👍